spec/integration/integration_spec.rb

```diff
@@ -1,18 +1,11 @@
 # frozen_string_literal: true
 
 require "spec_helper"
+require "que/worker" # required to prevent autoload races
 
 RSpec.describe "multiple workers" do
-  def with_workers(n)
-    workers = Array.new(n) { Que::Worker.new(wake_interval: 0.1) }
-    worker_threads = workers.map { |worker| Thread.new { worker.work_loop } }
-
-    yield
-
-    workers.each(&:stop!)
-    worker_threads.each do |thread|
-      raise "timed out waiting for worker to finish!" unless thread.join(5)
-    end
+  def with_workers(n, stop_timeout: 5)
```

Comment:
We can- and should!- repurpose these tests to target the `Worker.start_workers` code instead of our test rig. The only required addition is to cover our timeout case, which we do below.